optimum/habana/accelerate/utils/transformer_engine.py

```diff
@@ -0,0 +1,137 @@
+# coding=utf-8
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import torch
+from peft.tuners.lora.layer import Linear as PEFTLinear
+from optimum.habana.peft.layer import LoRALinear
+
+try:
+    import habana_frameworks.torch.hpex.experimental.transformer_engine as te
+    from habana_frameworks.torch.hpex.experimental.transformer_engine.distributed import activation_checkpointing
+
+    has_transformer_engine = True
+except ImportError:
+    has_transformer_engine = False
+
+
+def is_fp8_available():
+    return has_transformer_engine
+
+
+def _convert_model(model, to_transformer_engine=True, _convert_linear=True):
+    """
+    Recursively converts the linear and layernorm layers of a model to their `transformers_engine` counterpart.
+    """
+    if not is_fp8_available():
```

Comment:
Lets add support for SDPA also, can be ported from MLPerf branch. You can default it to disabled by setting _convert_sdpa=False for now. Later we will figure out a way to enable it only for Gaudi3.

Reply:
As discussed, will handle SDPA changes separately.